openmls/src/group/core_group/staged_commit.rs

```diff
@@ -329,6 +329,46 @@ pub struct StagedCommit {
 }
 
 impl StagedCommit {
+    /// Create a new [`StagedCommit`] from the provisional group state created
+    /// during the commit process.
+    pub(crate) fn from_provisional_state(
```

Comment:
I'm wondering if this split is really worth it at this point. The issues I see are:
 - The list of parameters is already above the clippy limit (generates a warning)
 - You re-calculate the serialized group context (you already did that in `create_commit()`)
 - The remaining code is just a few lines that you could also do in `create_commit()`

I think it would be better not to have this separate function for now and look at it again in the follow-up PR to see what the overlap with `stage_commit()` is.

Reply:
Good point about the clippy limit. I was doing it in `create_commit` beforehand, but @franziskuskiefer pointed out in PR #634 that this function is already too large, so I factored this out.